apps/api/src/app/controllers/resources/media-item.controller.ts

```diff
@@ -0,0 +1,37 @@
+import { Controller, Get, Param, Res } from '@nestjs/common';
+import { ApiOkResponse, ApiParam, ApiTags } from '@nestjs/swagger';
+import { MediaItemQueryService } from '../../../domain/services/media-item-query.service';
+import { resourceTypes } from '../../../domain/types/resourceTypes';
+import { isInternalError } from '../../../lib/errors/InternalError';
+import { isNotFound } from '../../../lib/types/not-found';
+import { MediaItemViewModel } from '../../../view-models/buildViewModelForResource/viewModels/media-item.view-model';
+import httpStatusCodes from '../../constants/httpStatusCodes';
+import { buildByIdApiParamMetadata, RESOURCES_ROUTE_PREFIX } from '../resourceViewModel.controller';
+import buildViewModelPathForResourceType from '../utilities/buildViewModelPathForResourceType';
+
+@ApiTags(RESOURCES_ROUTE_PREFIX)
+@Controller(
+    `${RESOURCES_ROUTE_PREFIX}/${buildViewModelPathForResourceType(resourceTypes.mediaItem)}`
+)
+export class MediaItemController {
+    constructor(private mediaItemQueryService: MediaItemQueryService) {}
+    @ApiParam(buildByIdApiParamMetadata())
+    @ApiOkResponse({ type: MediaItemViewModel })
+    @Get('/:id')
+    async fetchById(@Res() res, @Param('id') id: unknown) {
+        const searchResult = await this.mediaItemQueryService.fetchById(id);
+
+        if (isInternalError(searchResult))
+            return res.status(httpStatusCodes.badRequest).send(searchResult.toString());
+
+        if (isNotFound(searchResult) || !searchResult.published)
```

Comment:
Should filtering out the `unpublished` items be the responsibility of the service instead?

Reply:
It should be the responsibility of the controller to determine whether the request \ user should receive published items. The controller should then adapt this information to create arguments for the service call.

Reply:
done